flytekit/control_plane/workflow.py

```
@@ -0,0 +1,183 @@
+    def promote_from_model(
+        cls,
+        base_model: _workflow_models.WorkflowTemplate,
+        sub_workflows: Optional[Dict[_identifier.Identifier, _workflow_models.WorkflowTemplate]] = None,
+        tasks: Optional[Dict[_identifier.Identifier, _task_models.TaskTemplate]] = None,
+    ) -> "FlyteWorkflow":
+        base_model_non_system_nodes = cls.get_non_system_nodes(base_model.nodes)
+        sub_workflows = sub_workflows or {}
+        tasks = tasks or {}
+        node_map = {
+            n.id: _nodes.FlyteNode.promote_from_model(n, sub_workflows, tasks) for n in base_model_non_system_nodes
+        }
+
+        # Set upstream nodes for each node
+        for n in base_model_non_system_nodes:
+            current = node_map[n.id]
+            for upstream_id in current.upstream_node_ids:
+                upstream_node = node_map[upstream_id]
+                current << upstream_node
+
+        # No inputs/outputs specified, see the constructor for more information on the overrides.
+        return cls(
+            nodes=list(node_map.values()),
+            id=_identifier.Identifier.promote_from_model(base_model.id),
+            metadata=base_model.metadata,
+            metadata_defaults=base_model.metadata_defaults,
+            interface=_interfaces.TypedInterface.promote_from_model(base_model.interface),
+            output_bindings=base_model.outputs,
+        )
+
+    @_exception_scopes.system_entry_point
+    def register(self, project, domain, name, version):
+        # TODO
+        pass
+
+    @_exception_scopes.system_entry_point
+    def serialize(self):
+        # TODO
+        pass
+
+    @_exception_scopes.system_entry_point
+    def validate(self):
+        # TODO
+        pass
+
+    @_exception_scopes.system_entry_point
+    def create_launch_plan(self, *args, **kwargs):
+        # TODO
+        pass
+
+    @_exception_scopes.system_entry_point
+    def __call__(self, *arks, **input_map):
+        if len(args) > 0:
```

this doesn't work today right?  If it doesn't work can we just raise an error?